### DISCUSSION TITLE
Raise a window upon focused?

### DISCUSSION THREAD

**Discussion by mobsenpai** [hyprwm/Hyprland#5451] [Q&A] [answered]
Let's say I have to floating windows like in the pic attached and I want that the window I am focusing using movefocus from keyboard should come to top i.e gets raised. How will I do that?

Currently I have to click on the bottom window to make it come to top.

> attached pic
![image](https://github.com/hyprwm/Hyprland/assets/92603465/1d0b1ae0-ec1c-4e50-80cd-f1dc6f0a1e72)

> hyprland.conf on pastebin
https://pastebin.com/GkQYVVMV

> NOTE: this is nixos's output file so please ignore nix store links. It just targets to an executable.

**Comment by vaxerski** [comment]
you can make a small script that listens to the activewindow socket2 event and dispatches alterzorder

**Comment by 0xFMD** [comment]
```
bind = ALT, Tab, cyclenext,
bind = ALT, Tab, bringactivetotop

```

**Reply by vaxerski** [reply]
or that, I missed the "keyboard" part lol

**Reply by mobsenpai** [reply]
I don't want another keybind to do this. I want the window to come on top when i press super + hjkl.

**Reply by vaxerski** [reply]
this one keybind bud lmao

**Reply by 0xFMD** [reply]
try this
```
bind = super, h, bringactivetotop
bind = super, l, bringactivetotop
bind = super, k, bringactivetotop
bind = super, j, bringactivetotop

```

**Reply by mobsenpai** [reply]
But will it movefocus as well?

**Reply by 0xFMD** [reply]
yep

**Reply by vaxerski** [reply]
multiple binds to the same key will be executed in the order they were declared

**Reply by mobsenpai** [reply]
So I have to bind both movefocus and bringactivetotop to hjkl?

**Reply by 0xFMD** [reply]
yes

**Reply by mobsenpai** [reply]
Oh really! I will try it tonight and let you know then.

**Reply by mobsenpai** [reply]
Okay so that really worked. Now one last thing is that I saw that bringactivetotop is deprecated in the hyprland docs in favour of alterzorder is there any alternative way of doing this using alterzorder or will the bringactivetotop will get removed in future. Should I worry abt that or not?

**Reply by 0xFMD** [reply, marked as answer]
maybe this will work, bind the "_movefocus_" before it

```
bind = super, h, alterzorder, top
bind = super, l, alterzorder, top
bind = super, k, alterzorder, top
bind = super, j, alterzorder, top

```

**Reply by mobsenpai** [reply]
Okay that's it. Brilliantly worked. Thank you @MysticBytes786.

**Reply by Kommynct** [reply]
Weird, that doesn't work for me, it causes the focus to go back and forth rapidly

**Reply by mobsenpai** [reply]
You have to first define movefocus then after that alterzorder.

**Reply by Kommynct** [reply]
I have it setup like the following:

```
bind = ALT, s, movefocus, r
bind = ALT, h, movefocus, l
bind = ALT, t, movefocus, u
bind = ALT, n, movefocus, d
bind = ALT, s, alterzorder, top
bind = ALT, h, alterzorder, top
bind = ALT, t, alterzorder, top
bind = ALT, n, alterzorder, top
```

**Reply by mobsenpai** [reply]
Looks good to me. @MysticBytes786 any idea?

**Reply by Kommynct** [reply]
Upon further testing, alterzorder, top just does nothing when bound to a key for me, bringactivetotop works though.

**Comment by fiskhest** [comment]
I had a related yet unrelated problem on my end. If I have a window fullscreened and for example launch a new (browser) tab from that window, the original window would stay fullscreened but unfocused, requiring me to mouseclick original window, de-fullscreen, move focus to browser. I solved this with the following listener, which upon initial testing seems to work while not ruining the same workflow when initiated from a tiled window:
```
#!/usr/bin/env bash

function handle {
  if [[ ${1:0:6} == "urgent" ]]; then
      hyprctl dispatch focuscurrentorlast
      hyprctl dispatch fullscreen 0
      hyprctl dispatch focuscurrentorlast
      hyprctl dispatch fullscreen 0
  fi
}

socat - "UNIX-CONNECT:$XDG_RUNTIME_DIR/hypr/$HYPRLAND_INSTANCE_SIGNATURE/.socket2.sock" | while read -r line; do handle "$line" ; done
```

**Reply by klesh** [reply]
@fiskhest How did you auto start the script?
I wrote a similar script which works fine in the terminal, but it always fails when using `exec-once`

**Reply by fiskhest** [reply]
I unfortunately hit the same realization and never found a solution.

**Reply by klesh** [reply]
@fiskhest I might have a solution, check [the gist](https://gist.github.com/klesh/7b76f1f2694464db0f9acf6b42e54292)

**Reply by fiskhest** [reply]
@klesh thanks for sharing, I'll give it a whirl and update here afterwards.

**Reply by fiskhest** [reply]
Initial testing looks like a complete success 🎉 Huge thank you for sharing!! 🙏🏼
